Mfix DEM example and LSD model (#764)
lethe-cfd/lethe: doc/source/examples/dem/bouncing-particle/bouncing-particle.rst

Parameter file
--------------

Mesh
~~~~~

The ``grid type`` in this example is a ``hyper_cube``. Its dimensions are 2.0 m in every direction (from 0.0 m to 2.0 m), and since only one particle is used, the refinement is set to zero.

.. code-block:: text

    subsection mesh
      set type               = dealii
      set grid type          = hyper_cube
      set grid arguments     = 0.0 : 2.0 : false
      set initial refinement = 0
    end

Insertion info
~~~~~~~~~~~~~~~~~~~

Since the insertion of the particle must be done at as specific height, the ``list`` insertion method is being used. The ``insertion frequency`` can be set to any value, since we're only using one particle. The particle is at a height of 0.5 m in the center of the X,Y plane.

.. code-block:: text

    subsection insertion info
      set insertion method                               = list
      set insertion frequency                            = 10000
      set list x = 1.0
      set list y = 1.0
      set list z = 0.5
    end

Lagrangian physical properties
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> As mentioned earlier, different spring constant and restitution coefficient are being used through out this example. As you will soon realise, the spring constant is not defined explicitly in the parameter file, consult `DEM parameters <../../../theory/dem/dem.rst>`_ for more information about the definition of this constants.

Reviewer: ```suggestion
Different spring constant and restitution coefficient are used in this example. The spring constant is not defined explicitly in the parameter file, and, consequently, the physical properties must be chosen to obtain a desired spring constant. The interested reader can consult `DEM parameters <../../../theory/dem/dem.rst>`_ for more information about the definition of the sprint constant.
```

.. code-block:: text

  dem_3d bouncing_particle_XX.prm

Every simulation takes less than 30 seconds to run. A folder named according to the restitution coefficient used will be generated (``/out_xx``).

Post-processing
---------------
A Python post-processing code called ``Bouncing_particle_post_processing.py`` is provided with this example. It is used to compare the height reached by the particle at each bounces with the analytical solution of a hard sphere bouncing on a flat plane. This analytical solution consider instantaneous collision between the particle and the wall, thus the maximum height of each bounce can be express by the following expression:

.. math::
        h_{k}^{max} = (h_0 - r_p)e^{2k} + rp

with :math:`k` represent the :math:`k^{th}` bounce, :math:`h_0` the starting height and :math:`r_p` the radius of the particle.

Once the 6 simulations have been run, used the following line in your command line to run the post-processing code :

.. code-block:: text

  python3 Bouncing_particle_post_processing.py

> A figure will then appear comparing the expected values results with the analytical solution.

Reviewer: ```suggestion
A figure will be generated which compares the analytical solution with the simulation results.
```

Results and discussion
-----------------------
Animation of a bouncing particle with different restitution coefficient (:math:`K_n` = 5E6 N/m):

.. raw:: html

    <iframe width="560" height="315" src="https://www.youtube.com/embed/B1n776jHdoo" frameborder="0" allowfullscreen></iframe>

It can be seen that the particle with a restitution coefficient (:math:`e`) of 1.0 doesn't have any change in its rebounds. The other particles show a reduction in rebound height that seems to respect the analytical solution express earlier.

Now, using the post-processing code, it is possible to compare the effect of the normal spring constant of the conservation of the kinetic energy during the collision.

.. list-table::

    * - .. figure:: images/kn5E4.png
           :width: 260
           :alt: Mesh
           :align: center

>            Spring constant equal to 5E4.

Reviewer: The images are nice but it's not clear. You don't have symbols for the analytical solution and for the simulation results. Can you try to make a figure a bit clearer so we at least recognize that there are two symbols (or use a line for simulationa nd symbols for the analytical solution)
Must be clearer. It's very good, but requires a bit more work.

Author: I have symbol for the analytical solutions, right now it's small cross, but I understand that's not visible with the resolution used in the html. I'll change the marker types so that it's clearer.

Author: I'll ajust the current label in the legend so that we know ithey are the analytical.

Should I add the "numerical" label to the legend? I'm scare that there will be to many label in it...

      - .. figure:: images/kn5E5.png
           :width: 260
           :alt: Mesh
           :align: center

           Spring constant equal to 5E5.

      - .. figure:: images/kn5E6.png
           :width: 260
           :alt: Mesh
           :align: center

           Spring constant equal to 5E6.

> As it can be seen of the previous figures, when the normal spring constant becomes less stiff, the height of each bounce is less than what its suppose to be according to the analytical solution, except when the restitution coefficient is equal to 1.0. This is due to the assumption of instantaneous contact, which becomes false for an elastic particle. Since the particle is less stiff, the contact time between the particle and the wall is longer, thus the damping term in the force calculation comes into effect over a longer period and more kinetic energy is lost.

Reviewer: ```suggestion
As it can be seen of the previous figures, when the normal spring constant is lower, the height of each bounce is less than the analytical solution, except when the restitution coefficient is equal to 1.0. This is due to the assumption of instantaneous contact, which becomes false for a very elastic particle. In this case, the particle-wall overlap becomes significant compared to the particle diameter leading to additional energy dissipation. Essentially, particles which are overly soft break the underlying assumption of the soft sphere DEM model. The same conclusions were reached in the reference MFIX benchmark.
```

> 

Reviewer: Aa the stiffness is increased, the agreement between the results obtained in the simulations and the analytical solution improves.

A figure will then appear comparing the expected values results with the analytical solution.

Results and discussion
-----------------------
Animation of a bouncing particle with different restitution coefficient (:math:`K_n` = 5E6 N/m):

.. raw:: html

    <iframe width="560" height="315" src="https://www.youtube.com/embed/B1n776jHdoo" frameborder="0" allowfullscreen></iframe>

> It can be seen that the particle with a restitution coefficient (:math:`e`) of 1.0 doesn't have any change in its rebounds. The other particles show a reduction in rebound height that seems to respect the analytical solution express earlier.

Reviewer: ```suggestion
It can be seen that the particle with a restitution coefficient (:math:`e`) of 1.0 always rebounds at the same height. The other particles show a reduction in rebound height which follows the analytical solution expressed earlier.
```
